### PR TITLE
Show captain quick info dialog

### LINKED ISSUE #1849: Castle screen, right click on Captain's portrait
A right click and hold on the Captain's portrait could bring up an info window with the Captain's information (as for the hero).

In fheroes2:

Captain right click fh2 01

Other screen where it should also be possible to display the Captain's information window by right-clicking and holding on the Captain's portrait.

In fheroes2 (the captain's portrait is missing, it's a bug reported here: https://github.com/ihhub/fheroes2/issues/1848):

Captain right click fh2 02

### PR DESCRIPTION
**Pull request by ihhub**
close #1849

image

### CODE REVIEW

File: src/fheroes2/dialog/dialog_quickinfo.cpp [hunk 1]
```diff
@@ -868,15 +868,19 @@ void Dialog::QuickInfo( const Heroes & hero, const fheroes2::Rect & activeArea,
     std::string message;
 
     const Kingdom & kingdom = world.GetKingdom( conf.CurrentColor() );
-    const bool isFriend = hero.isFriends( conf.CurrentColor() );
+    const bool isFriend = ColorBase( hero.GetColor() ).isFriends( conf.CurrentColor() );
     const bool isUnderIdentifyHeroSpell = kingdom.Modes( Kingdom::IDENTIFYHERO );
     const bool showFullInfo = isFriend || isUnderIdentifyHeroSpell || kingdom.IsTileVisibleFromCrystalBall( hero.GetIndex() );
 
+    const Heroes * activeHero = dynamic_cast<const Heroes *>( &hero );
+    const Captain * activeCaptain = dynamic_cast<const Captain *>( &hero );
```
Comment (oleg-derevenetz) [inline src/fheroes2/dialog/dialog_quickinfo.cpp:876]: I'd add an assertion here to check that either the `activeHero` or the `activeCaptain` should be non-null.

File: src/fheroes2/castle/castle_dialog.cpp [hunk 2]
```diff
@@ -435,6 +435,9 @@ Castle::CastleDialogReturnValue Castle::OpenDialog( const bool readOnly, const b
             else if ( le.MousePressRight( buttonPrevCastle.area() ) ) {
                 Dialog::Message( _( "Show previous town" ), _( "Click to show previous town." ), Font::BIG );
             }
+            else if ( GetCaptain().isValid() && heroes.Guard() == nullptr && le.MousePressRight( rectSign1 ) ) {
```
Comment (oleg-derevenetz) [inline src/fheroes2/castle/castle_dialog.cpp:438]: AFAIK pressing the right mouse button over the portrait of a regular hero in the castle dialog does nothing. Why is this exception for captains?

**Review by oleg-derevenetz — COMMENTED** [review]
Hi @ihhub I left the one suggestion and one question here, please take a look when possible :)

File: src/fheroes2/castle/castle_dialog.cpp (hunk 2 above)
Comment (ihhub) [reply src/fheroes2/castle/castle_dialog.cpp:438]: I think you're right. It shouldn't be the case. I will revert it.

File: src/fheroes2/dialog/dialog_quickinfo.cpp (hunk 1 above)
Comment (ihhub) [reply src/fheroes2/dialog/dialog_quickinfo.cpp:876]: Done.

**Review by oleg-derevenetz — APPROVED** [review]

**Pull request merged**